metacat-main/src/main/java/com/netflix/metacat/main/configs/ServicesConfig.java

```diff
@@ -400,22 +401,25 @@ public MetadataService metadataService(
      * @param connectorManager     Connector manager to use
      * @param threadServiceManager Thread service manager to use
      * @param metacatThriftService Thrift service to use
+     * @param applicationContext the application context
+     *
      * @return The initialization service bean
      */
-    @Bean
+    @Bean(initMethod = "start", destroyMethod = "stop")
```

Comment:
This makes sure that the application context is ready before loading the catalogs. How do we make sure that this is initialized since none of the classes have a reference to it.

Reply:
Spring beans are eagerly initialized. So the bean itself should be initialized. As for the ordering, that I have only validated in practice that the controllers are initlialized before the traffic is allowed.

One surefire way to order this is ti inject the initializer to the V1 controllers - but let me look if there are more idiomatic options.

Reply:
Had to refactor the init service in Metacat. This was because the ApplicationRunner interface did not work as it was invoked after the Controllers and I was still getting 404s on startup. To make the order explicit, I needed to inject the initialization service to the controllers.

This also required me to split the init service to core and thrift init services to prevent a circular dependency.

Reply:
Can we use https://www.baeldung.com/spring-depends-on?

Reply:
We can. Your Spring-foo is good. Thanks. Updated.